src/test_lib_json/main.cpp

```diff
@@ -2499,6 +2499,17 @@ JSONTEST_FIXTURE(RValueTest, moveConstruction) {
 #endif
 }
 
+struct ErrorTest : JsonTest::TestCase {};
+JSONTEST_FIXTURE(ErrorTest, getErrorWorksFineWithStreams)
+{
+  const char TEST_VALUE[] = R"__({"X":3, "Y":oops})__";
+  JSONCPP_ISTRINGSTREAM ss(TEST_VALUE);
+  Json::Reader reader;
+  Json::Value result;
+  JSONTEST_ASSERT(!reader.parse(ss, result));
+  JSONTEST_ASSERT(reader.getFormattedErrorMessages().size());
```

Comment:
Should we also test the *content* of the formatted error messages?

Anyway, the real issue is whether pointers are correct. I'm not sure how to test that, since it's internal to the class.

Reply:
Yep, You are right, unfortunately, the only way is to look up through the internals of the class(downcasting via layout) and check it with /proc/self/maps or use _CrtIsValidHeapPointer or using VirtualQuery but all of such methods are messy.

Reply:
Yes, Determining the type of pointer in C++ requires a lot of mess(parsing /proc/self/maps, checking  regions via VirtualQuery, or using MSVC specific _CrtIsValidHeapPointer)